File: lib/backend/src/services/ProfileService.ts
import { v4 as uuidv4 } from 'uuid';
import { ConfigService } from './ConfigService';

export class ProfileService {
    static init() {
        const config = ConfigService.loadConfig();
        if (!config.profiles || Object.keys(config.profiles).length === 0) {
            this.migrateLegacyConfig(config);
        }
    }

    static migrateLegacyConfig(config: any) {
        const defaultProfileId = 'default';
        const now = new Date().toISOString();
        const defaultProfile = {
            id: defaultProfileId,
            name: 'Default',
            created: now,
            lastUsed: now,
            mods: config.installedMods || [],
            javaPath: config.javaPath || '',
            gameOptions: {
                minMemory: '1G',
                maxMemory: '4G',
                args: []
            }
        };

        ConfigService.saveConfig({
            profiles: { [defaultProfileId]: defaultProfile },
            activeProfileId: defaultProfileId
        });
    }

    static createProfile(name: string) {
        const id = uuidv4();
        const newProfile = {
            id,
            name,
            created: new Date().toISOString(),
            lastUsed: null,
            mods: [],
            javaPath: '',
            gameOptions: {
                minMemory: '1G',
                maxMemory: '4G',
                args: []
            }
        };
        
        const config = ConfigService.loadConfig();
        const profiles = config.profiles || {};
        profiles[id] = newProfile;
        
        ConfigService.saveConfig({ profiles });
        return newProfile;
    }

    static updateProfile(profileId: string, updates: any) {
        const config = ConfigService.loadConfig();
        const profiles = config.profiles || {};
        
        if (profiles[profileId]) {
            profiles[profileId] = { ...profiles[profileId], ...updates };
            ConfigService.saveConfig({ profiles });
            return profiles[profileId];
        }
        return null;
    }

    static getProfiles() {
        const config = ConfigService.loadConfig();
        return config.profiles || {};
    }

    static getActiveProfile() {
        const config = ConfigService.loadConfig();
        const activeId = config.activeProfileId || 'default';
        return config.profiles?.[activeId] || null;
    }
}

File: lib/backend/src/services/ProfileService.py
import uuid
import time
from datetime import datetime
from .ConfigService import ConfigService

class ProfileService:
    @classmethod
    def init(cls):
        config = ConfigService.load_config()
        if not config.get('profiles') or len(config.get('profiles', {})) == 0:
            cls.migrate_legacy_config(config)

    @classmethod
    def migrate_legacy_config(cls, config):
        default_profile_id = 'default'
        now = datetime.utcnow().isoformat() + "Z"
        
        default_profile = {
            "id": default_profile_id,
            "name": "Default",
            "created": now,
            "lastUsed": now,
            "mods": config.get('installedMods', []),
            "javaPath": config.get('javaPath', ''),
            "gameOptions": {
                "minMemory": "1G",
                "maxMemory": "4G",
                "args": []
            }
        }

        ConfigService.save_config({
            "profiles": {default_profile_id: default_profile},
            "activeProfileId": default_profile_id
        })

    @classmethod
    def create_profile(cls, name):
        profile_id = str(uuid.uuid4())
        now = datetime.utcnow().isoformat() + "Z"
        
        new_profile = {
            "id": profile_id,
            "name": name,
            "created": now,
            "lastUsed": None,
            "mods": [],
            "javaPath": "",
            "gameOptions": {
                "minMemory": "1G",
                "maxMemory": "4G",
                "args": []
            }
        }

        config = ConfigService.load_config()
        profiles = config.get('profiles', {})
        profiles[profile_id] = new_profile
        
        ConfigService.save_config({"profiles": profiles})
        return new_profile

    @classmethod
    def update_profile(cls, profile_id, updates):
        config = ConfigService.load_config()
        profiles = config.get('profiles', {})
        
        if profile_id in profiles:
            profiles[profile_id].update(updates)
            ConfigService.save_config({"profiles": profiles})
            return profiles[profile_id]
        
        return None

    @classmethod
    def get_profiles(cls):
        config = ConfigService.load_config()
        return config.get('profiles', {})

    @classmethod
    def get_active_profile(cls):
        config = ConfigService.load_config()
        active_id = config.get('activeProfileId', 'default')
        profiles = config.get('profiles', {})
        return profiles.get(active_id)
